CMakeLists.txt

```diff
@@ -156,71 +164,158 @@ if(UMF_BUILD_BENCHMARKS)
+        # files formatting.
+        set(CMAKE_FORMAT_REQUIRED "0.6")
+        if(NOT (CMAKE_FORMAT_VERSION VERSION_EQUAL CMAKE_FORMAT_REQUIRED))
+            message(FATAL_ERROR "Required cmake-format version is"
+                                "${CMAKE_FORMAT_REQUIRED}")
+        endif()
+    endif()
+
+    if(NOT CLANG_FORMAT AND NOT CMAKE_FORMAT)
+        message(
+            FATAL_ERROR
+                "UMF_FORMAT_CODE_STYLE=ON, but neither clang-format (required version: "
+                "${CLANG_FORMAT_REQUIRED}) nor cmake-format (required version: "
+                "${CMAKE_FORMAT_VERSION}) was found.")
     endif()
 
     # Obtain files for clang-format check
     set(format_glob)
-    foreach(DIR IN ITEMS include src test benchmark)
-        list(APPEND format_glob
+    foreach(
+        DIR IN
+        ITEMS include
+              examples
+              src
+              test
+              benchmark)
+        list(
+            APPEND
+            format_glob
             "${DIR}/*.h"
             "${DIR}/*.hpp"
             "${DIR}/*.c"
             "${DIR}/*.cpp"
             "${DIR}/**/*.h"
             "${DIR}/**/*.hpp"
             "${DIR}/**/*.c"
-            "${DIR}/**/*.cpp"
-        )
+            "${DIR}/**/*.cpp")
     endforeach()
     file(GLOB_RECURSE format_list ${format_glob})
 
-    message(STATUS "Adding clang-format-check and clang-format-apply make targets")
+    message(STATUS "Adding clang-format-check and clang-format-apply make "
+                   "targets")
 
-    add_custom_target(clang-format-check
-        COMMAND ${CLANG_FORMAT}
-            --style=file
-            --dry-run
-            -Werror
-            ${format_list}
+    add_custom_target(
+        clang-format-check
+        COMMAND ${CLANG_FORMAT} --style=file --dry-run -Werror ${format_list}
         COMMENT "Check files formatting using clang-format")
 
-    add_custom_target(clang-format-apply
-        COMMAND ${CLANG_FORMAT}
-            --style=file
-            --i
-            ${format_list}
+    add_custom_target(
+        clang-format-apply
+        COMMAND ${CLANG_FORMAT} --style=file --i ${format_list}
         COMMENT "Format files using clang-format")
+
+    # Obtain files for cmake-format check
+    set(format_cmake_glob)
+    foreach(
+        DIR IN
+        ITEMS examples
+              cmake
+              include
+              src
+              test
+              benchmark)
+        list(
+            APPEND
+            format_cmake_glob
+            "${DIR}/CMakeLists.txt"
+            "${DIR}/*.cmake"
+            "${DIR}/**/CMakeLists.txt"
+            "${DIR}/**/*.cmake")
+    endforeach()
+    file(GLOB_RECURSE format_cmake_list ${format_cmake_glob})
+    list(APPEND format_cmake_list "${PROJECT_SOURCE_DIR}/CMakeLists.txt")
+
+    # Print the list of files targeted for formatting by cmake-format
+    message(STATUS "Files to be formatted with cmake-format:")
```

Comment:
I believe you can delete it now.

IMHO, if this is to be printed always, it should be done in build time (within `make`), not as part of CMake configuration.

Reply:
I just deleted it here. We could add print in make for cmake-format and clang-format in other PR